### ISSUE TITLE
table_fuzzer out of memory issue

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5591]
I'm getting the following reliably when running `test/run interface.table_fuzzer` on `next`:
```
rethinkdb: Memory allocation failed. This usually means that we have run out of RAM. Aborting.
```

I compiled with recc++ in debug mode.

**Comment by danielmewes** [comment]
It looks like it's passing in release mode.

**Comment by danielmewes** [comment]
Got a backtrace for the allocation:
```
error: Error in src/errors.cc at line 33:
error: Memory
error: Backtrace:
error: Fri Apr  1 21:23:42 2016
       
       1 [0x20cdbeb]: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:108
       2 [0x16d7ab0]: backtrace_t::backtrace_t() at backtrace.cc:210 (discriminator 1)
       3 [0x16d8e6c]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:331
       4 [0x16d7737]: format_backtrace(bool) at backtrace.cc:205
       5 [0x20cd0fe]: report_fatal_error(char const*, int, char const*, ...) at errors.cc:86
       6 [0x20cce49]: crash_oom() at errors.cc:33
       7 [0x210cb6a]: raw_malloc_aligned(unsigned long, unsigned long) at utils.cc:269
       8 [0x210cbd5]: raw_malloc_page_aligned(unsigned long) at utils.cc:278
       9 [0x1500521]: scoped_alloc_t<char, &(raw_malloc_page_aligned(unsigned long)), &(raw_free_aligned(void*))>::scoped_alloc_t(unsigned long) at scoped.hpp:267
       10 [0x14fef66]: artificial_stack_t::artificial_stack_t(void (*)(), unsigned long) at context_switching.cc:48
       11 [0x1500ac2]: coro_t::coro_t() at coroutines.cc:145 (discriminator 2)
       12 [0x1501e56]: coro_t::get_coro() at coroutines.cc:402
       13 [0x1a90e61]: coro_t* coro_t::get_and_init_coro<std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const&, bool, auto_drainer_t::lock_t)> (table_query_client_t*, std::pair<peer_id_t, uuid_u>, table_query_bcard_t, bool, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const&, bool, auto_drainer_t::lock_t)> (table_query_client_t*, std::pair<peer_id_t, uuid_u>, table_query_bcard_t, bool, auto_drainer_t::lock_t)>&&) at coroutines.hpp:180
       14 [0x1a882fd]: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const&, bool, auto_drainer_t::lock_t)> (table_query_client_t*, std::pair<peer_id_t, uuid_u>, table_query_bcard_t, bool, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const&, bool, auto_drainer_t::lock_t)> (table_query_client_t*, std::pair<peer_id_t, uuid_u>, table_query_bcard_t, bool, auto_drainer_t::lock_t)>&&) at coroutines.hpp:63
       15 [0x1a80d7f]: table_query_client_t::update_registrant(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) at table_query_client.cc:491 (discriminator 1)
       16 [0x1aa116f]: void std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)>::operator()<std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, void>(table_query_client_t*, std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&) const at functional:601
       17 [0x1aa0f2b]: void std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> (table_query_client_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::__call<void, std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&, 0ul, 1ul, 2ul>(std::tuple<std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&>&&, std::_Index_tuple<0ul, 1ul, 2ul>) at functional:1295
       18 [0x1aa0e86]: void std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> (table_query_client_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::operator()<std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, void>(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&) at functional:1353
       19 [0x1aa0c27]: std::_Function_handler<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*), std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> (table_query_client_t*, std::_Placeholder<1>, std::_Placeholder<2>)> >::_M_invoke(std::_Any_data const&, std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) at functional:2073
       20 [0x19732e0]: std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)>::operator()(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) const at functional:2471
       21 [0x1972c41]: watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::read_all(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&) at watchable_map.tcc:202
       22 [0x197295b]: watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::all_subs_t::all_subs_t(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>*, std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&, initial_call_t) at watchable_map.tcc:16
       23 [0x1a80974]: table_query_client_t::table_query_client_t(uuid_u const&, mailbox_manager_t*, watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>*, multi_table_manager_t*, rdb_context_t*, table_meta_client_t*) at table_query_client.cc:32 (discriminator 6)
       24 [0x196e0c9]: namespace_repo_t::create_and_destroy_namespace_interface(namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t) at namespace_interface_repository.cc:132
       25 [0x197118c]: boost::_mfi::mf3<void, namespace_repo_t, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t>::operator()(namespace_repo_t*, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t) const at mem_fn_template.hpp:393 (discriminator 4)
       26 [0x1971077]: void boost::_bi::list4<boost::_bi::value<namespace_repo_t*>, boost::_bi::value<namespace_repo_t::namespace_cache_t*>, boost::_bi::value<uuid_u>, boost::_bi::value<auto_drainer_t::lock_t> >::operator()<boost::_mfi::mf3<void, namespace_repo_t, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t>, boost::_bi::list0>(boost::_bi::type<void>, boost::_mfi::mf3<void, namespace_repo_t, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t>&, boost::_bi::list0&, int) at bind.hpp:457
       27 [0x1970faf]: boost::_bi::bind_t<void, boost::_mfi::mf3<void, namespace_repo_t, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t>, boost::_bi::list4<boost::_bi::value<namespace_repo_t*>, boost::_bi::value<namespace_repo_t::namespace_cache_t*>, boost::_bi::value<uuid_u>, boost::_bi::value<auto_drainer_t::lock_t> > >::operator()() at bind_template.hpp:20
       28 [0x1970edc]: callable_action_instance_t<boost::_bi::bind_t<void, boost::_mfi::mf3<void, namespace_repo_t, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t>, boost::_bi::list4<boost::_bi::value<namespace_repo_t*>, boost::_bi::value<namespace_repo_t::namespace_cache_t*>, boost::_bi::value<uuid_u>, boost::_bi::value<auto_drainer_t::lock_t> > > >::run_action() at callable_action.hpp:32
       29 [0x150a2c6]: callable_action_wrapper_t::run() at runtime_utils.cc:48
       30 [0x1500f5d]: coro_t::run() at coroutines.cc:205 (discriminator 1)
       31 [0x196f79d]: coro_t* coro_t::spawn_sometime<boost::_bi::bind_t<void, boost::_mfi::mf3<void, namespace_repo_t, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t>, boost::_bi::list4<boost::_bi::value<namespace_repo_t*>, boost::_bi::value<namespace_repo_t::namespace_cache_t*>, boost::_bi::value<uuid_u>, boost::_bi::value<auto_drainer_t::lock_t> > > >(boost::_bi::bind_t<void, boost::_mfi::mf3<void, namespace_repo_t, namespace_repo_t::namespace_cache_t*, uuid_u const&, auto_drainer_t::lock_t>, boost::_bi::list4<boost::_bi::value<namespace_repo_t*>, boost::_bi::value<namespace_repo_t::namespace_cache_t*>, boost::_bi::value<uuid_u>, boost::_bi::value<auto_drainer_t::lock_t> > >&&) at coroutines.hpp:63
       32 [0x196e94c]: namespace_repo_t::get_namespace_interface(uuid_u const&, signal_t*) at namespace_interface_repository.cc:208 (discriminator 2)
       33 [0x1985190]: real_reql_cluster_interface_t::table_find(name_string_t const&, counted_t<ql::db_t const>, boost::optional<admin_identifier_format_t>, signal_t*, counted_t<base_table_t>*, admin_err_t*) at real_reql_cluster_interface.cc:465
       34 [0x1936365]: artificial_reql_cluster_interface_t::table_find(name_string_t const&, counted_t<ql::db_t const>, boost::optional<admin_identifier_format_t>, signal_t*, counted_t<base_table_t>*, admin_err_t*) at artificial_reql_cluster_interface.cc:165 (discriminator 1)
       35 [0x1ee47db]: ql::table_term_t::eval_impl(ql::scope_env_t*, ql::args_t*, ql::eval_flags_t) const at db_table.cc:873 (discriminator 3)
       36 [0x20ae6df]: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:220
       37 [0x1e50447]: ql::runtime_term_t::eval_on_current_stack(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:275
       38 [0x1e508ec]: /home/ssd3/daniel/rethinkdb/build/debug_clang/rethinkdb() [0x1e508ec] at 0x1e508ec ()
       39 [0x1e506a1]: /home/ssd3/daniel/rethinkdb/build/debug_clang/rethinkdb() [0x1e506a1] at 0x1e506a1 ()
       40 [0x1e5062b]: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:296
       41 [0x20ae905]: ql::op_term_t::maybe_grouped_data(ql::scope_env_t*, ql::argvec_t*, ql::eval_flags_t, counted_t<ql::grouped_data_t>*, scoped_ptr_t<ql::val_t>*) const at op.cc:300 (discriminator 1)
       42 [0x20ae1a0]: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:205
       43 [0x1e50447]: ql::runtime_term_t::eval_on_current_stack(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:275
       44 [0x1e508ec]: /home/ssd3/daniel/rethinkdb/build/debug_clang/rethinkdb() [0x1e508ec] at 0x1e508ec ()
       45 [0x1e506a1]: /home/ssd3/daniel/rethinkdb/build/debug_clang/rethinkdb() [0x1e506a1] at 0x1e506a1 ()
       46 [0x1e5062b]: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:296
       47 [0x1f1cee5]: ql::query_cache_t::ref_t::run(ql::env_t*, ql::response_t*) at query_cache.cc:271 (discriminator 1)
       48 [0x1f1c3dd]: ql::query_cache_t::ref_t::fill_response(ql::response_t*) at query_cache.cc:209
       49 [0x1e4c284]: rdb_query_server_t::run_query(ql::query_params_t*, ql::response_t*, signal_t*) at query_server.cc:45 (discriminator 1)
       50 [0x20ddbc5]: void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}::operator()() const::{lambda()#1}::operator()() const at server.cc:593
       51 [0x20dd7c3]: void save_exception<void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}::operator()() const::{lambda()#1}>(std::__exception_ptr::exception_ptr*, std::string*, cond_t*, void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}::operator()() const::{lambda()#1}&&) at server.cc:543
       52 [0x20dd4a5]: void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}::operator()() const at server.cc:591
       53 [0x20dd31c]: callable_action_instance_t<void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}>::run_action() at callable_action.hpp:32
       54 [0x150a2c6]: callable_action_wrapper_t::run() at runtime_utils.cc:48
       55 [0x1500f5d]: coro_t::run() at coroutines.cc:205 (discriminator 1)
       56 [0x20dd0fd]: void coro_t::spawn_now_dangerously<void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}>(void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}&&) at coroutines.hpp:57
       57 [0x20dc896]: void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*) at server.cc:581
       58 [0x20d8aa0]: query_server_t::handle_conn(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) at server.cc:453
       59 [0x20de4d0]: void std::_Mem_fn<void (query_server_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)>::operator()<scoped_ptr_t<linux_tcp_conn_descriptor_t>&, auto_drainer_t::lock_t&, void>(query_server_t*, scoped_ptr_t<linux_tcp_conn_descriptor_t>&, auto_drainer_t::lock_t&) const at functional:601
       60 [0x20de40b]: void std::_Bind<std::_Mem_fn<void (query_server_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (query_server_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)>::__call<void, scoped_ptr_t<linux_tcp_conn_descriptor_t>&, 0ul, 1ul, 2ul>(std::tuple<scoped_ptr_t<linux_tcp_conn_descriptor_t>&>&&, std::_Index_tuple<0ul, 1ul, 2ul>) at functional:1295
       61 [0x20de35d]: void std::_Bind<std::_Mem_fn<void (query_server_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (query_server_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)>::operator()<scoped_ptr_t<linux_tcp_conn_descriptor_t>&, void>(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:1353
       62 [0x20ddfe2]: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), std::_Bind<std::_Mem_fn<void (query_server_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (query_server_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)> >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:2073
       63 [0x14ed3d8]: std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)>::operator()(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) const at functional:2471
       64 [0x14e93ec]: linux_nonthrowing_tcp_listener_t::handle(int) at network.cc:1351
       65 [0x14efb99]: void std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)>::operator()<int&, void>(linux_nonthrowing_tcp_listener_t*, int&) const at functional:601
       66 [0x14efb00]: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1295
       67 [0x14efa86]: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::operator()<, void>() at functional:1353
       68 [0x14ef9fc]: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >::run_action() at callable_action.hpp:32
       69 [0x150a2c6]: callable_action_wrapper_t::run() at runtime_utils.cc:48
       70 [0x1500f5d]: coro_t::run() at coroutines.cc:205 (discriminator 1)
       71 [0x14ecffd]: void coro_t::spawn_now_dangerously<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>&&) at coroutines.hpp:57
       72 [0x14e82da]: linux_nonthrowing_tcp_listener_t::accept_loop(auto_drainer_t::lock_t) at network.cc:1318
       73 [0x14f121b]: void std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)>::operator()<auto_drainer_t::lock_t&, void>(linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t&) const at functional:601
       74 [0x14f1170]: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1295
       75 [0x14f10f6]: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::operator()<, void>() at functional:1353
       76 [0x14f103c]: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)> >::run_action() at callable_action.hpp:32
       77 [0x150a2c6]: callable_action_wrapper_t::run() at runtime_utils.cc:48
       78 [0x1500f5d]: coro_t::run() at coroutines.cc:205 (discriminator 1)
       79 [0x14ec3ad]: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>&&) at coroutines.hpp:63
       80 [0x14e7d22]: linux_nonthrowing_tcp_listener_t::begin_listening() at network.cc:1110 (discriminator 1)
       81 [0x14e9791]: linux_tcp_listener_t::linux_tcp_listener_t(std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int, std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)> const&) at network.cc:1382 (discriminator 1)
       82 [0x20d66b1]: query_server_t::query_server_t(rdb_context_t*, std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int, query_handler_t*, unsigned int, ssl_ctx_st*) at server.cc:203 (discriminator 1)
       83 [0x1e4c051]: rdb_query_server_t::rdb_query_server_t(std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, int, rdb_context_t*, server_config_client_t*, server_id_t const&, ssl_ctx_st*) at query_server.cc:22
       84 [0x175621d]: do_serve(io_backender_t*, bool, base_path_t const&, metadata_file_t*, serve_info_t const&, os_signal_cond_t*) at serve.cc:512
       85 [0x1757d5c]: serve(io_backender_t*, base_path_t const&, metadata_file_t*, serve_info_t const&, os_signal_cond_t*) at serve.cc:683
       86 [0x172c40a]: run_rethinkdb_serve(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, server_id_t const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*) at command_line.cc:1160
       87 [0x174260d]: void std::_Bind<void (*(base_path_t, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, server_id_t*, server_config_versioned_t*, cluster_semilattice_metadata_t*, directory_lock_t*, bool*))(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, server_id_t const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*)>::__call<void, , 0ul, 1ul, 2ul, 3ul, 4ul, 5ul, 6ul, 7ul, 8ul, 9ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul, 2ul, 3ul, 4ul, 5ul, 6ul, 7ul, 8ul, 9ul>) at functional:1295
       88 [0x17423e6]: void std::_Bind<void (*(base_path_t, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, server_id_t*, server_config_versioned_t*, cluster_semilattice_metadata_t*, directory_lock_t*, bool*))(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, server_id_t const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*)>::operator()<, void>() at functional:1353
       89 [0x1741ebd]: std::_Function_handler<void (), std::_Bind<void (*(base_path_t, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, server_id_t*, server_config_versioned_t*, cluster_semilattice_metadata_t*, directory_lock_t*, bool*))(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, server_id_t const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*)> >::_M_invoke(std::_Any_data const&) at functional:2073
       90 [0x14bd05e]: std::function<void ()>::operator()() const at functional:2471
       91 [0x14fd9e1]: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
       92 [0x14fe14a]: void std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)>::operator()<std::function<void ()>&, void>(starter_t*, std::function<void ()>&) const at functional:601
       93 [0x14fe0b0]: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1295
       94 [0x14fe036]: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::operator()<, void>() at functional:1353
error: Exiting.
```

**Comment by danielmewes** [comment]
I have a suspicion that we're hitting a kernel limit on the number of non-contiguous virtual memory regions due to our use of `mprotect` in the coroutine stacks. This is still very speculative though and I haven't fully verified this yet.

We can probably reduce the number of coroutines we allocate for the workload the table_fuzzer is generating.
However it's important for other workloads as well that we can handle large numbers of coroutines (e.g. users having a lot of changefeeds open at the same time). The limit can be increased on Linux by setting `/proc/sys/vm/max_map_count`, but it would be nice if we wouldn't require this.

**Comment by danielmewes** [comment]
A fix for this is in CR 3614, branch daniel_unprotect_coros.

**Comment by danielmewes** [comment]
Fixed in next by ee7da92 and v2.3.x by 9dc9b10.
